kubeflow/trainer/backends/container/adapters/podman.py

```
@@ -254,3 +253,29 @@ def get_network(self, network_id: str) -> Optional[dict]:
+            TimeoutError: If timeout is reached before container exits
+        """
+        try:
+            container = self.get_container(container_id)
+            result = container.wait(timeout=timeout)
+            # Podman wait() returns exit code directly
+            return int(result)
+        except Exception as e:
+            if "timeout" in str(e).lower():
+                raise TimeoutError(
+                    f"Container {container_id} did not exit within {timeout} seconds"
+                ) from e
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)